openupgrade_scripts/scripts/account_check_printing/14.0.1.0/upgrade_analysis_work.txt

```diff
@@ -0,0 +1,20 @@
+---Models in module 'account_check_printing'---
+---Fields in module 'account_check_printing'---
+account_check_printing / account.move             / preferred_payment_method_id (many2one): NEW relation: account.payment.method, isfunction: function, stored
+# DONE: pre-migration: created preferred_payment_method_id column and set the value
+
+account_check_printing / account.payment          / check_amount_in_words (char)  : now a function
+account_check_printing / account.payment          / check_number (char)           : now a function
+# NOTHING TO DO: automated field filling
+
+account_check_printing / res.company              / account_check_printing_layout (selection): selection_keys is now '['disabled']' ('['action_print_check_bottom', 'action_print_check_middle', 'action_print_check_top', 'disabled']')
```

Comment:
I'm not confortable with this one.
this field seems to be designed to be overloaded. The current migration script will reset the values, even if there are legit. In the other hand, I don't know where the others values are defined.
Not a clear point of view on this one. 
@pedrobaeza any insight ?

thanks.

Reply:
@legalsylvain 
With version 13 this field has 4 options and default is `action_print_check_top`
But with version 14 this field has only 1 option and default is `disabled`. I looked in the modules in version 14 and didn't see this module adding an option for this field.
Therefore, I think if I don't reset the values to `disabled` then upgrading will throw an error

Reply:
Yes, your proposal makes sense. I just don't understand the new odoo design.

We can merge it, and update later the script id OCA modules reintroduce missing feature.

Reply:
> Yes, your proposal makes sense. I just don't understand the new odoo design.
> 
> We can merge it, and update later the script if OCA modules reintroduce missing feature.

Thank bro @legalsylvain. I think you can merge it otherwise i think it will fail on migration